src/plugins/intel_gpu/src/plugin/ops/constant.cpp

```diff
@@ -71,6 +71,59 @@ struct ConstProperties {
     bool needsBatchInterpretation;
 };
 
+
+// Function to pack two 4-bit integers into one 8-bit integer
+inline uint8_t pack4to8(uint8_t s0, uint8_t s1) {
+    return (s1 << 4) | (s0 & 0x0F);
+}
+
+// Function to unpack an 8-bit integer into two 4-bit integers
+inline void unpack8to4(uint8_t v, uint8_t &v0, uint8_t &v1) {
+    v0 = v & 0x0F;
+    v1 = (v & 0xF0) >> 4;
+}
+
+// Function to convert a non-padded vector to a padded vector
+static void copy_to_padded_vector(const std::vector<int> &non_padded_dims, const char* non_padded_vec,
+                                                const std::vector<int> &padded_dims, char* padded_vec) {
+    size_t non_padded_dims_size = std::accumulate(non_padded_dims.begin(), non_padded_dims.end(), 1, std::multiplies<int>()) / 2;
+    size_t padded_dims_size = std::accumulate(padded_dims.begin(), padded_dims.end(), 1, std::multiplies<int>()) / 2;
+    size_t non_padded_row_size = non_padded_dims[1] / 2;
+    std::vector<uint8_t> temp_padding_buffer(padded_dims[1], 0);
+
+    for (int i = 0; i < non_padded_dims[0]; ++i) {
+        size_t padded_begin = i * padded_dims[1] / 2;
+        size_t non_padded_begin = i * non_padded_dims[1] / 2;
+        size_t padded_end = padded_begin + non_padded_row_size;
+        size_t non_padded_end = non_padded_begin + non_padded_row_size;
+
+        if (i % 2 == 0) {
+            std::memcpy(&padded_vec[padded_begin], &non_padded_vec[non_padded_begin], non_padded_row_size);
+            if (non_padded_end < non_padded_dims_size) {
+                uint8_t s0, s1;
+                unpack8to4(non_padded_vec[non_padded_end], s0, s1);
+                padded_vec[padded_end] = pack4to8(s0, 0);
+            }
+        } else {
+            for (size_t k = non_padded_begin, index = 0; k <= non_padded_end; ++k, ++index) {
+                uint8_t s0, s1;
+                unpack8to4(non_padded_vec[k], s0, s1);
+                if (k == non_padded_begin) {
+                    temp_padding_buffer[index] = s1;
+                } else {
+                    temp_padding_buffer[index] = s0;
+                    ++index;
+                    temp_padding_buffer[index] = s1;
+                }
```

Comment:
How long does it take? I'm afraid this will be very slow.  I think we need to find a better way.

Reply:
Optimized code, please review it again

Reply:
I made ocl kernel instead of CPU computation on constant creation. please review it again. thanks.